src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/vilkårperiode/AldersvilkårVurdering.kt

```diff
@@ -0,0 +1,93 @@
+    ): SvarJaNei {
+        val fødselsdato = grunnlagsData.grunnlag.fødsel?.fødselsdato
+
+        feilHvis(fødselsdato == null) { "Kan ikke vurdere aldersvilkår uten å vite fødselsdato til bruker" }
+
+        val gyldig: SvarJaNei =
+            when (vilkårperiode.type as MålgruppeType) {
+                MålgruppeType.AAP, MålgruppeType.NEDSATT_ARBEIDSEVNE, MålgruppeType.UFØRETRYGD ->
+                    vurderAldersvilkårForNedsattArbeidsevne(fødselsdato, vilkårperiode)
+                MålgruppeType.OMSTILLINGSSTØNAD -> vurderAldersvilkårForOmstillingsstønad(fødselsdato, vilkårperiode)
+                MålgruppeType.OVERGANGSSTØNAD -> SvarJaNei.JA_IMPLISITT
+                MålgruppeType.DAGPENGER -> SvarJaNei.NEI
+                MålgruppeType.SYKEPENGER_100_PROSENT -> SvarJaNei.NEI
+                MålgruppeType.INGEN_MÅLGRUPPE -> SvarJaNei.NEI
+            }
+
+        return gyldig
+    }
+
+    private fun heleVilkårsperiodenErFørBrukerFyller67År(
+        fødselsdato: LocalDate,
+        vilkårsperiodeFom: LocalDate,
+        vilkårsperiodeTom: LocalDate,
+    ): Boolean {
+        val sekstisyvÅrsDagenTilBruker = fødselsdato.plusYears(67)
+        feilHvis((vilkårsperiodeFom < sekstisyvÅrsDagenTilBruker) && (sekstisyvÅrsDagenTilBruker <= vilkårsperiodeTom)) {
+            "Brukeren fyller 67 år i løpet av vilkårsperioden"
+        }
+        return vilkårsperiodeTom < sekstisyvÅrsDagenTilBruker
+    }
+
+    private fun heleVilkårsperiodenErEtterBrukerFyller18År(
+        fødselsdato: LocalDate,
+        vilkårsperiodeFom: LocalDate,
+        vilkårsperiodeTom: LocalDate,
+    ): Boolean {
+        val attenårsdagenTilBruker = fødselsdato.plusYears(18)
+        feilHvis((vilkårsperiodeFom <= attenårsdagenTilBruker) && (attenårsdagenTilBruker <= vilkårsperiodeTom)) {
+            "Brukeren fyller 18 år i løpet av vilkårsperioden"
+        }
+        return attenårsdagenTilBruker < vilkårsperiodeFom
+    }
+
+    private fun vurderAldersvilkårForNedsattArbeidsevne(
+        fødselsdato: LocalDate,
+        vilkårperiode: LagreVilkårperiode,
+    ): SvarJaNei {
+        if (heleVilkårsperiodenErFørBrukerFyller67År(fødselsdato, vilkårperiode.fom, vilkårperiode.tom) &&
+            heleVilkårsperiodenErEtterBrukerFyller18År(fødselsdato, vilkårperiode.fom, vilkårperiode.tom)
+        ) {
+            return SvarJaNei.JA
+        }
+        return SvarJaNei.NEI
+    }
+
+    private fun vurderAldersvilkårForOmstillingsstønad(
+        fødselsdato: LocalDate,
+        vilkårperiode: LagreVilkårperiode,
+    ): SvarJaNei =
+        if (heleVilkårsperiodenErFørBrukerFyller67År(
+                fødselsdato,
+                vilkårperiode.fom,
+                vilkårperiode.tom,
+            )
+        ) {
+            SvarJaNei.JA
+        } else {
+            SvarJaNei.NEI
+        }
+
+    data class VurderingFaktaEtterlevelseAldersvilkår(
+        val vilkårperiodeFom: LocalDate,
+        val vilkårperiodeTom: LocalDate,
```

Comment:
Hva er behovet for `vilkårperiodeFom` og `vilkårperiodeTom` her?

Reply:
Jeg la disse inn igjen etter diskusjonen vi hadde på stand-up, hvor vi ble enige om at alle input-parametre som inngikk i en automatisk vurdering skulle lagres sammen som JSON. FOM og TOM er allerede lagret på Vilkårsperioden, så dersom du syns dette er overkill så kan jeg fjerne de herfra

Reply:
Fjernet: [ffe087b](https://github.com/navikt/tilleggsstonader-sak/pull/586/commits/ffe087b8528f2c3ceb1f12b0261f420e0d94de4c)